test/distributed/test_c10d_nccl.py

```diff
@@ -203,29 +204,45 @@ def test_init_no_gpus(self):
             c10d.ProcessGroupNCCL(store, self.rank, self.world_size)
 
 
-class ProcessGroupNCCLTest(TestCase):
-    MAIN_PROCESS_RANK = 0
+class ProcessGroupNCCLTest(MultiProcessTestCase):
+    def _create_process_group_nccl(self, store, rank, world_size, opts):
+        # create nccl pg with opts
+        pg = c10d.ProcessGroupNCCL(store, self.rank, self.world_size, opts)
+        return pg
 
-    def setUp(self):
-        self.rank = self.MAIN_PROCESS_RANK
-        self.world_size = 1
-        self.file = tempfile.NamedTemporaryFile(delete=False)
+    def opts(self, high_priority_stream=False):
+        opts = c10d.ProcessGroupNCCL.Options()
+        opts.is_high_priority_stream = high_priority_stream
+        return opts
 
+    def setUp(self):
+        super(ProcessGroupNCCLTest, self).setUp()
         # NCCL_BLOCKING_WAIT overrides NCCL_ASYNC_ERROR_HANDLING hence tests
         # that use NCCL_BLOCKING_WAIT will test it as expected.
         os.environ["NCCL_ASYNC_ERROR_HANDLING"] = "1"
-        self.num_gpus = torch.cuda.device_count()
+        # self.num_gpus = torch.cuda.device_count()
+        self._spawn_processes()
 
     def tearDown(self):
-        pass
+        super(ProcessGroupNCCLTest, self).tearDown()
+        try:
+            os.remove(self.file_name)
+        except OSError:
+            pass
+
+    @property
+    def world_size(self):
+        return 2
 
     @requires_nccl()
     @sandcastle_skip_if(torch.cuda.device_count() < 2, "NCCL test requires 2+ GPUs")
     def test_empty_tensors(self):
-        store = c10d.FileStore(self.file.name, self.world_size)
-        pg = c10d.ProcessGroupNCCL(store, self.rank, self.world_size)
+        store = c10d.FileStore(self.file_name, self.world_size)
+        pg = self._create_process_group_nccl(store, self.rank, self.world_size, self.opts())
+        rank_to_GPU = init_multigpu_helper(self.world_size, "nccl")
+        local_device_idx = rank_to_GPU[self.rank][0]
```

Comment:
This code seems to be very repetitive in each test. Maybe we can put them somewhere inside an object?